app/src/main/java/co/kr/bemyplan/ui/purchase/after/adapter/DailyContentsAdapter.kt

```diff
@@ -1,38 +1,97 @@
 import android.view.ViewGroup
+import androidx.databinding.BindingAdapter
+import androidx.databinding.ObservableArrayList
+import androidx.databinding.ViewDataBinding
 import androidx.recyclerview.widget.RecyclerView
-import co.kr.bemyplan.data.DailyContents
 import co.kr.bemyplan.data.Spot
-import co.kr.bemyplan.databinding.DailyContentsListBinding
+import co.kr.bemyplan.databinding.ItemDailyContentsBinding
+import co.kr.bemyplan.databinding.ItemDailyRouteBinding
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+import com.bumptech.glide.request.RequestOptions
+import java.lang.IllegalStateException
 
-class DailyContentsAdapter: RecyclerView.Adapter<DailyContentsAdapter.DailyContentsViewHolder>() {
+class DailyContentsAdapter(private val viewType: Int): RecyclerView.Adapter<DailyContentsAdapter.DailyContentsViewHolder>() {
 
-    private val dailyContentsList = mutableListOf<DailyContents>()
-    private val spotList = mutableListOf<Spot>()
+    private var spotList = listOf<Spot>()
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DailyContentsViewHolder {
-        val binding = DailyContentsListBinding.inflate(
-            LayoutInflater.from(parent.context),
-            parent, false
-        )
+    override fun getItemViewType(position: Int) = viewType
 
-        return DailyContentsViewHolder(binding)
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = when(viewType) {
+        TYPE_CONTENTS -> {
+            val binding = ItemDailyContentsBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent, false
+            )
+            ContentsViewHolder(binding)
+        }
+        TYPE_ROUTE -> {
+            val binding = ItemDailyRouteBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent, false
+            )
+            RouteViewHolder(binding)
+        }
+        else -> {
+            throw IllegalStateException("Not Found ViewHolder Type $viewType")
+        }
     }
 
     override fun onBindViewHolder(holder: DailyContentsViewHolder, position: Int) {
-        holder.onBind(dailyContentsList[position])
+        when(holder) {
+            is ContentsViewHolder -> {
+                holder.onBind(spotList[position])
+            }
+            is RouteViewHolder -> {
+                holder.onBind(spotList[position])
+            }
+        }
     }
 
-    override fun getItemCount() = dailyContentsList.size
+    override fun getItemCount() = spotList.size
+
+    @SuppressLint("NotifyDataSetChanged")
+    fun setItems(items: List<Spot>) {
+        spotList = items
+        notifyDataSetChanged()
+    }
+
+    // MVVM을 위한 코드
+   @BindingAdapter("bind:spotList")
+    fun bindSpotList(recyclerView: RecyclerView, items: ObservableArrayList<Spot>) {
+        val adapter = recyclerView.adapter as? DailyContentsAdapter
+        adapter?.setItems(items)
+    }
```

Comment:
Jetpack 컴포넌트 중에 LiveData를 사용할 것 같은데..

ObservableArrayList를 사용하지 않구 LiveData + List 형태가 동일한 기능을 하는데 그걸 기준으로 BindingAdapter를 만들어도 괜찮지않을까용?!👀👀

Reply:
사실 그리고 BindingAdapter의 경우
```suggestion
        fun RecyclerView.bindSpotList(items: ObservableArrayList<Spot>) {
         (adapter as? DailyContentsAdapter).setItems(items)
     }
```
요런식으로 확장함수 형식으로도 만들 수 있답니당 ~.~

Reply:
사실 **BindingAdapter**의 경우 앱 전역에서 static하게 선언되서 모든 바인딩 레이아웃에서 접근이 가능하기때문에 이렇게 RecyclerAdapter 내부에 정의하기보다는 object를 따로 파서 보통 거기서 관리하는 편입니당~!!

Reply:
게다가 이부분은 조금 우려되서 하는 말이 있다면.. LiveData로 만약 바꿀 생각이라면.. 나는 해당 리사이클러뷰가 사용되는 액티비티나 프래그먼트 코드단에서 데이터를 **observe**해서 데이터를 집어 넣어줄 것 같으..! 그 이유는 LiveData의 경우 뷰의 라이프사이클을 관측하면서 그게 변할때마다 항상 최신의 데이터(아니면 데이터 값이 변경되면 또 그것도 적용시켜서 뷰에 전달해줌 ㅇㅇ)를 반환하는 식으로 동작한단말이야?? 

근데 바인딩 어댑터에서 라이브데이터를 활용하려고 하면 **LifeCycleOwner**를 전달해주기가 어렵기때문에.. observe는 못하고 끽해야 LiveData의 밸류에 접근해서 값을 받아오는 식으로 밖에 코드를 못 짤 것 같단말이지?? 근데 그렇게 라이브데이터를 사용할 거면 사실 사용하는 의미가 없기때문에.. 이 데이터를 집어넣어주는 부분은 뷰 단으로 빼주는 게 어떤 지 조심스레 제안해봅니다..

Reply:
> 사실 **BindingAdapter**의 경우 앱 전역에서 static하게 선언되서 모든 바인딩 레이아웃에서 접근이 가능하기때문에 이렇게 RecyclerAdapter 내부에 정의하기보다는 object를 따로 파서 보통 거기서 관리하는 편입니당~!!

아 그게 di인가 보군요...

Reply:
> Jetpack 컴포넌트 중에 LiveData를 사용할 것 같은데..
> 
> ObservableArrayList를 사용하지 않구 LiveData + List 형태가 동일한 기능을 하는데 그걸 기준으로 BindingAdapter를 만들어도 괜찮지않을까용?!👀👀

사실 이 부분에 있어서 개념정리가 덜 됐습니다. 구글링하다가 대부분 저렇게 적용을 하길래 이해하지 못한 상태로 넣은 코드입니다. 나중에 이해하게 될 때, 다시 코멘트 주신 거 보겠습니다.